relays/client-bridge-hub-rococo/src/runtime_wrapper.rs

```diff
@@ -69,3 +69,75 @@ impl From<UtilityCall<Call>> for Call {
 		Call::Utility(call)
 	}
 }
+
+// TODO: remove this and use common from cumulus-like once fixed (https://github.com/paritytech/parity-bridges-common/issues/1598)
+/// Module with rewarding bridge signed extension support
+pub mod rewarding_bridge_signed_extension {
+	use bp_polkadot_core::{Balance, Hash, Nonce, PolkadotLike};
+	use bp_runtime::extensions::*;
+
+	type RewardingBridgeSignedExtra = (
+		CheckNonZeroSender,
+		CheckSpecVersion,
+		CheckTxVersion,
+		CheckGenesis<PolkadotLike>,
+		CheckEra<PolkadotLike>,
+		CheckNonce<Nonce>,
+		CheckWeight,
+		ChargeTransactionPayment<PolkadotLike>,
+		BridgeRejectObsoleteHeadersAndMessages,
+		RefundBridgedParachainMessages,
+		RefundBridgedParachainMessages,
```

Comment:
Sorry for going back on this

I think it would make sense to group the 2 refund extensions together. At least in the actual runtime. Something like:

```
RewardingBridgeSignedExtra = (
		CheckNonZeroSender,
		CheckSpecVersion,
		CheckTxVersion,
		CheckGenesis<PolkadotLike>,
		CheckEra<PolkadotLike>,
		CheckNonce<Nonce>,
		CheckWeight,
		ChargeTransactionPayment<PolkadotLike>,
		BridgeRejectObsoleteHeadersAndMessages,
		(RefundBridgedParachainMessages, RefundBridgedParachainMessages)
	);
```

But, even more generic, thinking again about https://github.com/paritytech/parity-bridges-common/pull/1894#discussion_r1112041653 , here I think it would be better to rename `RefundBridgedParachainMessages` to `RefundBridgedParachainMessagesTuple` (or something clearer, maybe even add a comment) in order to be able to use it in any indirect runtime as a wildcard for either a single `RefundBridgedParachainMessages` or a tuple of `(RefundBridgedParachainMessages, RefundBridgedParachainMessages, ...)`

Reply:
and how should runtime's side look like?

```
pub type SignedExtra = (
	frame_system::CheckNonZeroSender<Runtime>,
	frame_system::CheckSpecVersion<Runtime>,
	frame_system::CheckTxVersion<Runtime>,
	frame_system::CheckGenesis<Runtime>,
	frame_system::CheckEra<Runtime>,
	frame_system::CheckNonce<Runtime>,
	frame_system::CheckWeight<Runtime>,
	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
	BridgeRejectObsoleteHeadersAndMessages,
	(BridgeRefundBridgeHubRococoRelayers, BridgeRefundBridgeHubRococoRelayers),
);
```

`SignedExtension` has some tuple impl:

```
impl<AccountId, Call: Dispatchable> SignedExtension for Tuple {
	for_tuples!( where #( Tuple: SignedExtension<AccountId=AccountId, Call=Call,> )* );
	type AccountId = AccountId;
	type Call = Call;
	const IDENTIFIER: &'static str = "You should call `identifier()`!";
	for_tuples!( type AdditionalSigned = ( #( Tuple::AdditionalSigned ),* ); );
	for_tuples!( type Pre = ( #( Tuple::Pre ),* ); );
```

Reply:
> and how should runtime's side look like?

Yes, like that, exactly

```
pub type SignedExtra = (
	frame_system::CheckNonZeroSender<Runtime>,
	frame_system::CheckSpecVersion<Runtime>,
	frame_system::CheckTxVersion<Runtime>,
	frame_system::CheckGenesis<Runtime>,
	frame_system::CheckEra<Runtime>,
	frame_system::CheckNonce<Runtime>,
	frame_system::CheckWeight<Runtime>,
	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
	BridgeRejectObsoleteHeadersAndMessages,
	(BridgeRefundBridgeHubRococoRelayers, BridgeRefundBridgeHubWococoRelayers),
);
```

I tried something like this with 2 test refund signed extensions for millau and it worked.

> SignedExtension has some tuple impl:

Yes, it works for nested tuples.

Reply:
@serban300 
something like this: https://github.com/paritytech/parity-bridges-common/pull/1894/commits/83ad36081decd09fa229f41ba81e08597283ac67 ?

I havent tried it yet with runing runtime (tomorrow), but at least it compiles :)

Reply:
No, no, sorry, I'm bad at explaining. Sorry for misleading you. This would work, but we don't need this level of complexity. What I meant is on the relayer side you can just leave it as it was before:

```
/// The `SignedExtensionSchema` for `RefundBridgedParachainMessages`.
pub type RefundBridgedParachainMessages = GenericSignedExtensionSchema<(), ()>;
```

```
type RewardingBridgeSignedExtra = (
		CheckNonZeroSender,
		CheckSpecVersion,
		CheckTxVersion,
		CheckGenesis<PolkadotLike>,
		CheckEra<PolkadotLike>,
		CheckNonce<Nonce>,
		CheckWeight,
		ChargeTransactionPayment<PolkadotLike>,
		BridgeRejectObsoleteHeadersAndMessages,
		RefundRelayerForMessagesFromParachain,
		RefundBridgedParachainMessages,
	);
```

and just rename `RefundBridgedParachainMessages` to something like `RefundBridgedParachainMessagesTuple` (or a better name) and add a comment to clarify that this signed extension schema can be used as a counterpart for a `RefundBridgedParachainMessages` or a tuple with as many `(RefundBridgedParachainMessages, RefundBridgedParachainMessages, ...)` in the actual runtime. Because on the relay side, `()` works even if in the actual runtime we have `()` or `((), ())`, or `((), (), ())` ... . The transaction will work because it will decode to the same thing anyway.

Reply:
@serban300 
hehe, ok, I got it now what you meant, it is like I said [here](https://github.com/paritytech/parity-bridges-common/pull/1894/files#r1112065075) that we dont even need to add anything, because after `ChargeTransactionPayment<PolkadotLike>` we have `BridgeRejectObsoleteHeadersAndMessages` which is `()` so it covers everything `()` or `((), ())`, or `((), (), ())`

now I understand what you meant by `RefundBridgedParachainMessages` as Wildcard/placeholder which relies on the scale encoding for `()` or `((), ())`, or `((), (), ())` is the same

yeah, my commit is opposite, and can be totally explicit: `rewarding_bridge_signed_extension::RewardingBridgeSignedExtension<((), ()), ((), ())>,`

now we now several ways, but for simplicity I am going with your placeholder-like way

Reply:
Looks good now !

Reply:
well, [as we found out](https://github.com/paritytech/cumulus/pull/2194#discussion_r1113203831), downside of this wildcard/placeholder way is, that we cannot use this for testing `check_additional_signed` (with [complex way](https://github.com/paritytech/parity-bridges-common/pull/1894#discussion_r1112132383) it would be possible )

I had to remove `check_additional_signed` from cumulus side
https://github.com/paritytech/cumulus/pull/2194/files#diff-554a991cc27b681ef4c6491717f9d505b167ca7bf9e5490f88b8532754d70de9R1164-R1202

----------

anyway, regardless of wildcard/placeholder or complex way, this **Cumulus test** [ensure_signed_extension_definition_is_compatible_with_relay](https://github.com/paritytech/cumulus/pull/2194/files#diff-554a991cc27b681ef4c6491717f9d505b167ca7bf9e5490f88b8532754d70de9R1164) 
just checks Runtime's SignedExtension vs. definition in subtree `bridges/primitives/chain-bridge-hub-cumulus`

so there are several implications:
1. **Cumulus test** is kind of useful:
- it means, if we change Runtime's definition, then we need to also change subtree `bridges/primitives/chain-bridge-hub-cumulus`
- which means somebody has to backport/PR that change to the Bridges repo, but this can be released/merged only after new Runtime is onboarded onlive
2. `pub mod rewarding_bridge_signed_extension` does not belong to the  `primitives/chain-bridge-hub-cumulus` but should be moved to `relays/*` layer, e.g. `relays/client-cumulus` (because it is used just by relayer)
- if we do this, we lose ability to do that **Cumulus test**, because we dont propage `relays/*` layer to Cumulus subtree
-   but sure still we could make exception in `verify-pallets-build.sh` for `relays/client-cumulus` and use it as `[dev-dependencies]`
- it is just a question of what is more clear

Reply:
Could you open a separate issue related to this points please ? In order to not lose track of them

Reply:
ok, I just put it as a part https://github.com/paritytech/parity-bridges-common/issues/1896